### DISCUSSION TITLE
In-process communication instead of over the network?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [ergo-services/ergo#185] [General]
Could this also be used as a model for in-process communication between actors? So, a bit like the reactive concept?

**Comment by halturin** [comment]
Didnt get your question, could you please clarify it? what do you mean "in-process"?

**Comment by Robert-M-Muench** [comment]
"An actor-based Framework with **network transparency** for creating event-driven architecture in Golang."

I understand the main use case to be building distributed (actors running on different hardware or VM nodes) applications, and my question is about using this to build an actor-based application running on one machine.

**Reply by halturin** [reply]
Sure, you can start single node and spawn there as many processes as you want. You may also start multiple nodes on a single host. 

See "call" example https://github.com/ergo-services/examples

**Comment by Robert-M-Muench** [comment]
Ok, being even more specific: Will that work for in-process communication, or is it too heavy?

One binary, running on one machine, using several threads. So, like an additional abstraction layer on top of Go channels.

**Reply by halturin** [reply]
Process in ergo is an abstraction on top of the goroutine https://docs.ergo.services/basics/process
